### ISSUE TITLE
Wikimedia Commons filenames with special characters in data-background-image

### ISSUE THREAD

**Issue by p3palazzo** [hakimel/reveal.js#2831]
The `data-background-image` field does not seem to like some Wikimedia Commons URLs that have special characters in them, such as this one: https://upload.wikimedia.org/wikipedia/commons/b/bd/Grand_dessein_d%E2%80%99Henri_IV_%28ch%C3%A2teau_du_Louvre%29%2C_vers_1600-1615.jpg

I tried both with the encoded strings and with non-encoded strings. This link works fine as an image URL *on* the slide, though. Can anyone else reproduce it, and are there any thoughts on how to work around this issue?

**Comment by ChrisLasar** [comment]
Same here with filenames containing a german umlaut (ä,ö,ü)

**Comment by p3palazzo** [comment]
On a second pass, it seems I was letting some URL-encoded sequences slip through or making errors in copying the unencoded characters. Reveal.js seems to be handling the **unencoded** addresses just fine.

**Comment by madalu** [comment]
I can confirm that this issue occurs with URL-encoded sequences in data-background-image. It does not occur if the urls are unencoded/unescaped.

**Comment by p3palazzo** [comment]
> I can confirm that this issue occurs with URL-encoded sequences in data-background-image. It does not occur if the urls are unencoded/unescaped.

Indeed, but I still have trouble with URLs/filenames that contain commas `,` — for some reason the URL is truncated at the comma, even if it is unencoded.

**Comment by hakimel** [comment]
I can reproduce the issue with the Wikimedia URL you included but haven't found a way to fix it yet. We run `encodeURI` for all background images here: https://github.com/hakimel/reveal.js/blob/master/js/controllers/slidecontent.js#L106. I tried decoding the URL before encoding it but wasn't able to find a way to get it to work. 

Commas are used to delimit multiple background images so they won't work in URLs. Support for multiple backgrounds was merged recently in #2940.
